snapcraft/internal/lifecycle.py

```diff
@@ -368,6 +368,9 @@ def _snap_data_from_dir(directory):
 
 
 def snap(project_options, directory=None, output=None):
+    # Check for our prerequesite external command early
+    repo.check_for_command('mksquashfs')
```

Comment:
This might become rather complex very easily, but what about something like ubuntu_packages['squashfs-tools'] and respective options for other platforms?